runAna5DFAPathGeneration_algorithm.py

```diff
@@ -0,0 +1,227 @@
+ Connects to AvaFrame
+ Generated by Plugin Builder: http://g-sherman.github.io/Qgis-Plugin-Builder/
+                              -------------------
+        begin                : 2021-08-26
+        copyright            : (C) 2021 by AvaFrame Team
+        email                : felix@avaframe.org
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+
+__author__ = 'AvaFrame Team'
+__date__ = '2022'
+__copyright__ = '(C) 2022 by AvaFrame Team'
+
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
+
+import subprocess
+
+from qgis.PyQt.QtCore import QCoreApplication
+from qgis.core import (QgsProcessing,
+                       QgsProcessingException,
+                       QgsProcessingAlgorithm,
+                       QgsProcessingParameterFeatureSource,
+                       QgsProcessingParameterRasterLayer,
+                       QgsProcessingParameterMultipleLayers,
+                       QgsProcessingParameterFolderDestination,
+                       QgsProcessingOutputVectorLayer,
+                       QgsProcessingParameterBoolean,
+                       QgsProcessingOutputMultipleLayers
+                       )
+
+
+class runAna5DFAPathGenerationAlgorithm(QgsProcessingAlgorithm):
+    """
+    This is the AvaFrame Connection, i.e. the part running with QGis. For this
+    connector to work, more installation is needed. See instructions at docs.avaframe.org
+    """
+
+    DEM = 'DEM'
+    REL = 'REL'
+    RUNDFA = 'RUNDFA'
+    FOLDEST = 'FOLDEST'
+    OUTPUT = 'OUTPUT'
+    OUTPPR = 'OUTPPR'
+
+    def initAlgorithm(self, config):
+        """
+        Here we define the inputs and output of the algorithm, along
+        with some other properties.
+        """
+
+        self.addParameter(QgsProcessingParameterRasterLayer(
+            self.DEM,
+            self.tr("DEM layer")))
+
+        self.addParameter(QgsProcessingParameterMultipleLayers(
+            self.REL,
+            self.tr('Release layer(s)'),
+            layerType=QgsProcessing.TypeVectorAnyGeometry
+        ))
+
+        self.addParameter(QgsProcessingParameterBoolean(
+            self.RUNDFA,
+            self.tr('Run dense flow (com1DFA)'),
+            optional=True,
+            defaultValue=True
+        ))
```

Comment:
This needs to be removed. (The option can then be removed lower down as well)

If the user only specifies DEM and REL and OUTPUT Dir, there is no way to specify a directory with existing simulations! 

So if a user wants to do this, they have to use the script variant of AvaFrame...